test/puppetlabs/puppetdb/testutils/services.clj

```diff
@@ -352,3 +355,28 @@
   (until-consumed
    (fn []
      (pdb-client/submit-command-via-http! base-url cmd version payload))))
+
+(defn wait-for-server-processing
+  "Returns a truthy value indicating whether the wait was
+  successful (i.e. didn't time out).  Given a false or nil timeout,
+  waits forever."
+  ([server] (wait-for-server-processing server nil))
```

Comment:
Some of this seems unnecessary. I don't see any usage of the 1 arg version. With that gone I think the code below becomes simpler.